source/server-discovery-and-monitoring/tests/integration/pool-cleared-error.yml

```diff
@@ -0,0 +1,197 @@
+# Test SDAM error handling.
+runOn:
+    # This test requires retryable writes and failCommand appName.
+    - minServerVersion: "4.4"
+      topology: ["replicaset", "sharded"]
+
+database_name: &database_name "sdam-tests"
+collection_name: &collection_name "pool-cleared-error"
+
+data: []
+
+tests:
+  - description: PoolClearedError does not mark server unknown
+    clientOptions:
+      retryWrites: true
+      maxPoolSize: 1
+      appname: poolClearedErrorTest
+    operations:
+      # Perform an operation to ensure the node is discovered.
+      - name: insertOne
+        object: collection
+        arguments:
+          document:
+            _id: 1
+      # Configure the next insert to fail with a network error which will
+      # clear the pool leaving it paused until the server is rediscovered.
+      - name: configureFailPoint
+        object: testRunner
+        arguments:
+          failPoint:
+            configureFailPoint: failCommand
+            mode: { times: 1 }
+            data:
+                failCommands: ["insert"]
+                closeConnection: true
+                appName: poolClearedErrorTest
+      # Start threads.
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread1
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread2
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread3
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread4
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread5
+      - name: startThread
+        object: testRunner
+        arguments:
+          name: thread6
+      # Perform concurrent insert operations. The first one to execute will
+      # fail with a network error, mark the server Unknown, clear the pool,
+      # and retry.
+      # The other operations will either:
+      # - Notice the pool is paused, fail with a PoolClearedError, and retry.
+      # - Or block waiting in server selection until the server is rediscoverd.
+      #
+      # Note that this test does not gaurantee that a PoolClearedError will be
```

Comment:
We could guarantee this by adding a `blockTimeMS` to the failpoint, which should give enough time for the other threads to enter the wait queue.

Also, a few minor typos here "gaurantee" -> "guarantee", "rediscoverd" -> "rediscovered"